### ISSUE TITLE
force_ssl_in_redirect_uri does not work for multiple uris when urn:ietf:wg:oauth:2.0:oob is in list

### ISSUE THREAD

**Issue by bradymholt** [doorkeeper-gem/doorkeeper#1057]
### Steps to reproduce

With `force_ssl_in_redirect_uri !Rails.env.development?` set in config/initializers/dookeeper.rb, non ssl URLs appear to be allowed if `urn:ietf:wg:oauth:2.0:oob` is in the list.

This spec fails:

```
it 'prevents insecure redirect uris when multiple URIs are specified' do
  Doorkeeper.configure do
    force_ssl_in_redirect_uri true
  end

  app = create(:oauth_application, user_id: user.id)
  app.redirect_uri = "urn:ietf:wg:oauth:2.0:oob\n" +
                     "http://www.scarydomain.com"
  app.validate

  expect(app.errors[:redirect_uri]).to include('must be an HTTPS/SSL URI.')
end
```

### Expected behavior

I expect `app.errors[:redirect_uri]` in the spec above to contain  "must be an HTTPS/SSL URI.".

### Actual behavior

The above spec fails

### System configuration

**Ruby version**: 2.4.3

**Gemfile.lock**: doorkeeper (4.2.0)

**Comment by nbulaj** [comment]
Already solved in #976 

Update Doorkeeper to 4.3.1

**Comment by bradymholt** [comment]
Thank you!
